### PR TITLE
Add comments to why TaskType can be a physical property.

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
Add comments to why TaskType can be a physical property.

### What is changed and how it works?
Just try to figure out when will a TaskType can be physical property and how does TaskType works? We do this by debuging a certain SQL like “ explain select count(*) from tableName group by colName;”

### Check List 

Tests 

 - No code

Code changes

 - Has added some comments

Side effects

 - None

Related changes

 - None

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8735)

### CODE REVIEW

File: planner/core/exhaust_physical_plans.go [hunk 1]
```diff
@@ -815,8 +814,10 @@ func (la *LogicalAggregation) getStreamAggs(prop *property.PhysicalProperty) []P
 		// The table read of "CopDoubleReadTaskType" can't promises the sort
 		// property that the stream aggregation required, no need to consider.
 		for _, taskTp := range []property.TaskType{property.CopSingleReadTaskType, property.RootTaskType} {
+			//Add taskTp in childrenReqProps, it will pass to it's child in findBestTask
```
Comment (eurekaka) [inline planner/core/exhaust_physical_plans.go]: Please add space between `//` and first word, and add `.` in the end of the sentence.

File: planner/core/find_best_task.go [hunk 2]
```diff
@@ -112,6 +112,10 @@ func (p *baseLogicalPlan) findBestTask(prop *property.PhysicalProperty) (bestTas
 		// Next, get the bestTask with enforced prop
 		prop.Cols = []*expression.Column{}
 	}
+	//
```
Comment (eurekaka) [inline planner/core/find_best_task.go]: Please remove this line, you can reference existing comment format in the code base.

File: planner/core/task.go [hunk 3]
```diff
@@ -487,6 +487,10 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 func (p *PhysicalHashAgg) attach2Task(tasks ...task) task {
 	cardinality := p.statsInfo().RowCount
 	t := tasks[0].copy()
+	/*
+	 *	Since we may get a cop task from child (identified in child props), because we wanna push agg down to cop.
+	 *  So we need a copTask here to add partialAgg in the physical plan and finally convert it to a rootTask
+	 */
```
Comment (eurekaka) [inline planner/core/task.go]: use `//` please.

File: planner/core/rule_eliminate_projection.go [hunk 4]
```diff
@@ -70,6 +70,7 @@ func resolveExprAndReplace(origin expression.Expression, replace map[string]*exp
 	}
 }
 
+//may be we can spare the idle physicalPlan into free pool rather than throwing it to gc
```
Comment (winoros) [inline planner/core/rule_eliminate_projection.go]: Such comment is not appropriate as the comment of function header.
You can open an issue and then discuss with us.

File: planner/core/task.go [hunk 5]
```diff
@@ -499,6 +503,7 @@ func (p *PhysicalHashAgg) attach2Task(tasks ...task) task {
 				cop.indexPlan = partialAgg
 			}
 		}
+		//Finally finish the cop task and return the RootTask
```
Comment (winoros) [inline planner/core/task.go]: The comment above the `if block` is enough.

**Comment by XuHuaiyu** [comment]
This is PR is duplicate with #8742 , we may close this PR.
You can make new changes in the local branch `mybranch`, and push the new commit to the corresponding branch in your remote repository.

**Comment by winoros** [comment]
closed for duplicated pr

**Pull request closed without merging**